apacheconfig/wloader.py

```diff
@@ -30,98 +28,149 @@ class Node():
 
     @abc.abstractmethod
     def __str__(self):
-        raise NotImplementedError()
+        """Writes this node to a raw string. To get more metadata about the
+        object, use ``repr``.
+        """
+        pass
+
+    @abc.abstractmethod
+    def __repr__(self):
+        """Returns a string containing object metadata."""
+        pass
 
     @property
-    def type(self):
+    def parser_type(self):
         """A typestring as defined by the apacheconfig parser.
 
-        A single *Node class can have multiple possible values for this. For
-        instance, ItemNode is a generic representation of any single directive
-        but has a couple different possible `type`s, including `comment`,
-        `statement`, and `include`-- depending on which the caller may
-        decide to treat ItemNode differently.
+        The possible values for this are:
+
+        :class:`apacheconfig.ItemNode`: ``comment``, ``statement``,
+        ``include``, ``includeoptional``
+
+        :returns: The typestring (as defined by the apacheconfig parser) for
+                  this node.
+        :rtype: `str`
         """
         if self._type is None:
             raise NotImplementedError()
         return self._type
 
     @property
     def whitespace(self):
-        """Trailing or preceding whitespace.
-
-        Each Item or BlockNode keeps track of the whitespace preceding it.
-        For the first element in the configuration file, there could be no
-        whitespace preceding it at all, in which case this should return the
-        empty string.
-
-        ContentsNode is special in that it keeps track of the trailing white-
-        space.
-        """
+        """A string representing literal trailing or preceding whitespace
+        for this node. Can be overwritten.
+
+        Each ``ItemNode`` or ``BlockNode`` keeps track of the whitespace
+        preceding it. For the first element in the configuration file, there
+        could be no whitespace preceding it at all, in which case this should
+        return the empty string.
+
+        ContentsNode is special in that it keeps track of the *trailing*
+        whitespace. For example::
+
+            ItemNode('\\n  option value').whitespace => "\\n  "
+            ItemNode('\\n  # comment').whitespace => "\\n  "
+            BlockNode('\\n  <a>\\n</a>').whitespace => "\\n  "
+            ContentsNode('\\n  option value # comment\\n').whitespace => "\\n"
+
+        :returns: a string containing literal preceding or trailing whitespace
+                  information for this node.
+        :rtype: `str`
+       """
         if self._whitespace is None:
             raise NotImplementedError()
         return self._whitespace
 
     @whitespace.setter
     def whitespace(self, value):
+        """A string representing literal trailing or preceding whitespace
```

Comment:
I think Sphinx wants us to have the first line of the docstring short and sweet. We are advised to expand on what's in title in the next paragraph.